analysis/dataset_clean/fn-ref.R

```diff
@@ -0,0 +1,87 @@
+  
+  input$sub_cat_covidhospital <- ordered(input$sub_cat_covidhospital, 
+                                            levels = c("non_hospitalised",
+                                                       "hospitalised",
+                                                       "no_infection"))
+  
+  # Set reference level for variable: cov_cat_ethnicity --------------------------
+  print('Set reference level for variable: cov_cat_ethnicity')
+  
+  levels(input$cov_cat_ethnicity) <- list("Missing" = "0", "White" = "1", 
+                                          "Mixed" = "2", "South Asian" = "3", 
+                                          "Black" = "4", "Other" = "5")
+  
+  input$cov_cat_ethnicity <- ordered(input$cov_cat_ethnicity, 
+                                     levels = c("White","Mixed",
+                                                "South Asian","Black",
+                                                "Other","Missing"))
+  
+  # Set reference level for variable: cov_cat_imd -------------------------------
+  print('Set reference level for variable: cov_cat_imd')
+  
+  input$cov_cat_imd <- ordered(input$cov_cat_imd, 
+                               levels = c("1 (most deprived)","2","3","4","5 (least deprived)"))
+  
+  # Set reference level for variable: strat_cat_region -----------------------------
+  print('Set reference level for variable: strat_cat_region')
+
+  input$strat_cat_region <- factor(input$strat_cat_region,  
+                                   levels = c("East", 
+                                              "East Midlands", "London", 
+                                              "North East", "North West", 
+                                              "South East", "South West", 
+                                              "West Midlands", "Yorkshire and The Humber"))
+  input$strat_cat_region <- relevel(input$strat_cat_region, ref = "East")
+  
+  # Set reference level for variable: cov_cat_smoking ---------------------
+  print('Set reference level for variable: cov_cat_smoking')
+  
+  levels(input$cov_cat_smoking) <- list("Ever smoker" = "E", "Missing" = "M", "Never smoker" = "N", "Current smoker" = "S")
+  
+  input$cov_cat_smoking <- ordered(input$cov_cat_smoking, levels = c("Never smoker","Ever smoker","Current smoker","Missing"))
+  
+  # Set reference level for variable: cov_cat_sex --------------------------------
+  print('Set reference level for variable: cov_cat_sex')
+  
+  levels(input$cov_cat_sex) <- list("Female" = "female", "Male" = "male")
+  input$cov_cat_sex <- factor(input$cov_cat_sex, 
+                              levels = c("Female", "Male", "Unknown"))
+  input$cov_cat_sex <- relevel(input$cov_cat_sex, ref = "Female")
+
+  # Set reference level for variable: vax_cat_jcvi_group -------------------------
+  print('Set reference level for variable: vax_cat_jcvi_group')
+  
+  input$vax_cat_jcvi_group <- ordered(input$vax_cat_jcvi_group, 
```

Comment:
Individuals with JCVI group 99 should have been removed now that the ref function is at the end of the script.